src/main/java/com/matt/forgehax/mods/DerpMod.java

```diff
@@ -0,0 +1,244 @@
+
+    sendRotatePacket(player, player.rotationPitch, player.rotationYawHead);
+  }
+
+  private CPacketEntityAction.Action actionFromSneaking(boolean sneaking) {
+    return sneaking
+        ? CPacketEntityAction.Action.START_SNEAKING
+        : CPacketEntityAction.Action.STOP_SNEAKING;
+  }
+
+  private void sendSneakPacket(EntityPlayer player, boolean sneaking) {
+    this.sneaking = sneaking;
+    ignoreAndSend(new CPacketEntityAction(player, actionFromSneaking(sneaking)));
+  }
+
+  private void sneakDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendSneakPacket(player, player.isSneaking());
+  }
+
+  @Override
+  protected void onDisabled() {
+    if (rotate.get()) rotateDisabled();
+    if (sneak.get()) sneakDisabled();
+  }
+
+  private void sendHitPacket(EnumHand hand) {
+    ignoreAndSend(new CPacketAnimation(hand));
+  }
+
+  private boolean chance(float chance) {
+    return chance == 1f || (chance != 0f && Math.random() <= chance);
+  }
+
+  @SubscribeEvent
+  public void onLocalPlayerUpdate(LocalPlayerUpdateEvent event) {
+    EntityPlayer player = (EntityPlayer) event.getEntityLiving();
+    if (player == null) return;
+    Random rng = player.getRNG();
+
+    float effectiveSpeed = speed.get();
+
+    if (timerSync.get() && timerMod != null && timerMod.isEnabled())
+      effectiveSpeed /= timerMod.factor.get();
+
+    error += effectiveSpeed;
+    int iter = (int) error;
+    error -= iter;
+
+    for (int i = 0; i < iter; i++) {
+      if (rotate.get() && chance(rotateChance.get()))
+        sendRotatePacket(player, rng.nextInt(180) - 90, rng.nextInt(360) - 180);
+
+      if (sneak.get() && chance(sneakChance.get()))
+        sendSneakPacket(player, !sneaking);
+
+      if (hit.get() && chance(hitChance.get()))
+        sendHitPacket(chance(hitPan.get()) ? EnumHand.OFF_HAND : EnumHand.MAIN_HAND);
+    }
+  }
+
+  @SubscribeEvent
+  public void onPacketSending(PacketEvent.Outgoing.Pre event) {
+    Packet<?> packet = event.getPacket();
+    if (isIgnored(packet)) return;
+
+    if (rotate.get() && (packet instanceof CPacketPlayer.Rotation || packet instanceof CPacketPlayer.PositionRotation)) {
+      event.setCanceled(true);
+
+      if (packet instanceof CPacketPlayer.PositionRotation) {
+        ignoreAndSend(new CPacketPlayer.Position(
+            ((CPacketPlayer) packet).getX(0),
+            ((CPacketPlayer) packet).getY(0),
+            ((CPacketPlayer) packet).getZ(0),
+            ((CPacketPlayer) packet).isOnGround()
+        ));
+      }
+    } else if (sneak.get() && packet instanceof CPacketEntityAction) {
+      CPacketEntityAction.Action action = ((CPacketEntityAction) packet).getAction();
+
+      if (action == CPacketEntityAction.Action.START_SNEAKING || action == CPacketEntityAction.Action.STOP_SNEAKING)
+        sneaking = action == CPacketEntityAction.Action.START_SNEAKING;
+    } else if (hit.get() && packet instanceof CPacketAnimation)
```

Comment:
This is for arm swinging, correct? If so, I believe blocking this packet from sending will cause anti cheats to flag the player.

Reply:
Only when hit is turned on, and it sends all sorts of hit packets anyway.

Reply:
Derp is not meant to bypass anticheats anyway, it's not a serious mod